aeolus/aeolus.cpp

```diff
@@ -52,32 +52,46 @@ Synthesizer* createAeolus()
 //   Aeolus
 //---------------------------------------------------------
 
-Aeolus::Aeolus()
+Aeolus::Aeolus() : Synthesizer()
       {
-      model = 0;
+      model = nullptr;
       patchList.append(new MidiPatch { false, "Aeolus", 0, 0, 0, "Aeolus" });
 
       _sc_cmode = 0;
       _sc_group = 0;
       _running = false;
+      _hold = 0;
       _nplay = 0;
-      _fsamp = 0;
       _nasect = 0;
       _ndivis = 0;
+      _revsize = 0.0f;
+      _revtime = 0.0f;
       nout = 0;
-      _ifc_init = 0;
-      for (int i = 0; i < NGROUP; i++)
-            _ifelms [i] = 0;
+      _fsamp = 0.0f;
+      _fsize = 0;
+      _ifc_init = nullptr;
+      memset(_midimap, 0, sizeof(_midimap));
+      memset(_asectp, 0, sizeof(_asectp));
+      memset(_divisp, 0, sizeof(_divisp));
       memset(_keymap, 0, sizeof(_keymap));
+      memset(_audiopar, 0, sizeof(_audiopar));
+      memset(routb, 0, sizeof(routb));
+      memset(loutb, 0, sizeof(loutb));
+      memset(_asectpar, 0, sizeof(_asectpar));
+      memset(_ifelms, 0, sizeof(_ifelms));
+      memset(_tempstr, 0, sizeof(_tempstr));
       }
 
 Aeolus::~Aeolus()
       {
-      delete model;
+      if (model != nullptr)
```

Comment:
Simply use `if (model)` here. Same for the below two.

Reply:
Simply don't use `if` at all, see http://www.cs.technion.ac.il/users/yechiel/c++-faq/delete-handles-null.html, it is superfluos.

Reply:
But the author said:

> After fixing that, I experienced crashes in the aeolus destructor, because of uninitialized fields.

Reply:
delete something uninitialized may crash, but won't get caught by checking against nullptr

Reply:
My C roots are showing.  You are right; the real fix was to make sure that every field of the object was initialized.  I have removed the `if` statements.

Reply:
Well, `free(NULL)` in C is also guaranteed to work ;-)